installer/scripts/service_control

```diff
@@ -16,49 +19,164 @@
-        if [ -f $ETC_DIR/$1/conf/omsadmin.conf ]; then
-            . "$ETC_DIR/$1/conf/omsadmin.conf"
-        else
-            echo "Warning: Specified workspace $1 does not exist on the machine. Check the correctness of the workspace ID and that onboarding succeeded."
-            exit 0
-        fi
+        initial_conf_dir="$ETC_DIR/$1/conf/omsadmin.conf"
+    fi
+
+    WS_STATUS=0
+    check_omsadmin_conf $initial_conf_dir
+    WS_STATUS=$?
+    if [ "$WS_STATUS" -eq "0" ]; then
+        . $initial_conf_dir
+        WS_STATUS=$?
     fi
 
-    VAR_DIR_WS=$VAR_DIR/$WORKSPACE_ID
-    ETC_DIR_WS=$ETC_DIR/$WORKSPACE_ID
+    if [ "$WS_STATUS" -eq "0" ]; then
 
-    TMP_DIR=$VAR_DIR_WS/tmp
-    STATE_DIR=$VAR_DIR_WS/state
-    RUN_DIR=$VAR_DIR_WS/run
-    LOG_DIR=$VAR_DIR_WS/log
-    CERT_DIR=$ETC_DIR_WS/certs
-    CONF_DIR=$ETC_DIR_WS/conf
+        VAR_DIR_WS=$VAR_DIR/$WORKSPACE_ID
+        ETC_DIR_WS=$ETC_DIR/$WORKSPACE_ID
 
-    LOGFILE=$LOG_DIR/omsagent.log
-    PIDFILE=$RUN_DIR/omsagent.pid
-    CONFFILE=$CONF_DIR/omsagent.conf
-    OMSADMIN_CONF=$CONF_DIR/omsadmin.conf
+        TMP_DIR=$VAR_DIR_WS/tmp
+        STATE_DIR=$VAR_DIR_WS/state
+        RUN_DIR=$VAR_DIR_WS/run
+        LOG_DIR=$VAR_DIR_WS/log
+        CERT_DIR=$ETC_DIR_WS/certs
+        CONF_DIR=$ETC_DIR_WS/conf
 
-    OMSAGENT_WS=omsagent-$WORKSPACE_ID
+        LOGFILE=$LOG_DIR/omsagent.log
+        PIDFILE=$RUN_DIR/omsagent.pid
+        CONFFILE=$CONF_DIR/omsagent.conf
+        OMSADMIN_CONF=$CONF_DIR/omsadmin.conf
+
+        OMSAGENT_WS=omsagent-$WORKSPACE_ID
+    fi
+    return $WS_STATUS
+}
+
+check_omsadmin_conf()
+{
+    # Checks for a series of identifier assignment lines which should always be present
+    # for an onboarded workspace to be complete and healthy.
+    local _fs=$1
+    local ec=0
+    VALIDATION_SUPPLEMENT_MESSAGE="Onboarding incorrect or incomplete."
+    if [ -f $_fs ]; then
+        if ! check_file_has_line_pattern $_fs "WORKSPACE_ID=$REGEX_UUID";       then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "AGENT_GUID=$REGEX_UUID";         then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "LOG_FACILITY=.*";                then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "CERTIFICATE_UPDATE_ENDPOINT=.*"; then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "URL_TLD=.*";                     then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "DSC_ENDPOINT=.*";                then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "OMS_ENDPOINT=.*";                then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "AZURE_RESOURCE_ID=.*";           then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "OMSCLOUD_ID=.*";                 then ec=$(expr $ec + 1); fi
+        if ! check_file_has_line_pattern $_fs "UUID=$REGEX_UUID";               then ec=$(expr $ec + 1); fi
```

Comment:
As per above I'm making them all in higher number categories.  I'm not paying attention to the value consistency with other scripts for now, as that looks like a lot of work, but that is a reasonable long term ideal.  See next push.

Reply:
The better way is to have a little routine that does those pieces.  That is a little higher maintenence in shell since there are no embedded routines, but it's easy then to test the thing.  I suggest we wait until we separate out the api from the main and at that time, we can go through and divide things out for better unit testing anyway.